library/core/src/macros/mod.rs

```diff
@@ -1508,26 +1508,34 @@ pub(crate) mod builtin {
 
     /// Unstable implementation detail of the `rustc` compiler, do not use.
     #[rustc_builtin_macro]
-    #[stable(feature = "rust1", since = "1.0.0")]
     #[allow_internal_unstable(core_intrinsics, libstd_sys_internals)]
     #[rustc_deprecated(
         since = "1.52.0",
         reason = "rustc-serialize is deprecated and no longer supported"
     )]
     #[doc(hidden)] // While technically stable, using it is unstable, and deprecated. Hide it.
+    #[unstable(
+        feature = "rustc_encodable_decodable",
+        issue = "none",
+        reason = "RustcDecodable cannot be used stably"
+    )]
     pub macro RustcDecodable($item:item) {
         /* compiler built-in */
     }
 
     /// Unstable implementation detail of the `rustc` compiler, do not use.
     #[rustc_builtin_macro]
-    #[stable(feature = "rust1", since = "1.0.0")]
     #[allow_internal_unstable(core_intrinsics)]
     #[rustc_deprecated(
         since = "1.52.0",
         reason = "rustc-serialize is deprecated and no longer supported"
     )]
     #[doc(hidden)] // While technically stable, using it is unstable, and deprecated. Hide it.
+    #[unstable(
+        feature = "rustc_encodable_decodable",
+        issue = "none",
+        reason = "RustcDecodable cannot be used stably"
+    )]
```

Comment:
You can mark both macros as soft-unstable (see `macro bench` for an example), then the instability will be reported as a deny-by-default lint and won't break other crates depending on crates using these macros.